### ISSUE TITLE
User suggestion:  poor set of magnetic field lines

### ISSUE THREAD

**Issue by oliver-phet** [phetsims/faradays-law#155]
> Quick question – the Faraday’s Law HTML5 interactive has a rather poor set of magnetic field lines that, if drawn by our pupils, would not see them score full marks in an exam. I have attached a screen shot. It’s a fairly minor thing, but the closeness of the field lines at the centre has them touching. As you go out from the ends of the magnet, the closeness of the field lines is highly uneven.
>![PhEt bad mag field 2](https://user-images.githubusercontent.com/5090275/74549264-91a60580-4f0c-11ea-8bff-e5ccfa8d97b5.jpg)
>  
> Having just done a Google search for images of field lines around bar magnets, it seems that this sort of thing is a very common error!
>  
> [This is reasonable](http://3.bp.blogspot.com/-d5SG2djnY8A/VOlpcVbKGqI/AAAAAAAAD_s/DMuht-6R-3g/s1600/magnetic-fields-around-magnet.PNG): 
> ![](http://3.bp.blogspot.com/-d5SG2djnY8A/VOlpcVbKGqI/AAAAAAAAD_s/DMuht-6R-3g/s1600/magnetic-fields-around-magnet.PNG)
> but not quite right at the poles, as some lines are much closer for no reason  (other than artistic).
> 
>  
> [This](https://image.slidesharecdn.com/6magneticfieldlines-100714171208-phpapp01/95/6-magnetic-field-lines-4-728.jpg?cb=1279127595) is dreadful as three lines merge to one: 
![image](https://user-images.githubusercontent.com/5090275/74549393-c9ad4880-4f0c-11ea-98e0-29db2e339e0c.png)
> 
>  
> [This is brilliant](https://ds055uzetaobb.cloudfront.net/image_optimizer/1aab95c7d944fadde68041d352e380c71edb312e.png) (as the site name suggests!) - 
![](https://ds055uzetaobb.cloudfront.net/image_optimizer/1aab95c7d944fadde68041d352e380c71edb312e.png)
 found here: https://brilliant.org/wiki/magnetic-field-lines/ although arguably some of the field lines are not quite uniformly spaced near the poles.
>  
> I wonder if this is something you could look into please? As it’s only an on/off interactive element, it should be an easy fix to replace the bitmap used?

**Comment by jbphet** [comment]
This doesn't look like it was ever assigned to anyone, and this sim is about to be re-released, so I think a call should be made on whether we want to make in updates to the field lines.  Assigning to @ariel-phet for consideration.

**Comment by jbphet** [comment]
Adding @arouinfar as and assignee, since she too has been involved in the design.

**Comment by ariel-phet** [comment]
@jbphet - the teacher does bring up some valid points. I think I have never paid much attention since I think the field lines were drawn to match the flash sim of MD.

However, considering we pride ourselves on fairly accurate physics, the user suggestion should be addressed in my opinion. I think something closer to the image shown below would be good, withe the following points:

1. evenly spaced lines that do not intersect (appear to come to a point) at the poles (suggesting a uniform field)
2. An "evenly growing" spacing that suggests a more uniform decrease in field strength as you move out from the poles
3. The stroke width of the lines might need to be decreased to accommodate them not looking like they emanate from the same point at the poles. 

![magnetic-field-lines-around-a-bar-magnet---teachoo-(1)-(1)](https://user-images.githubusercontent.com/4051174/97378591-4c521a00-1888-11eb-8822-069b0b55fc79.jpg)

**Comment by ariel-phet** [comment]
Marking high priority since I would naively think this should be a fairly easy fix, although it may take some time to get looking nice, it more seems tweaky than fundamentally difficult (please correct me if that is wrong).

**Comment by arouinfar** [comment]
I agree with @ariel-phet. I think we should try redrawing the magnetic field lines to look more accurate.

**Comment by ariel-phet** [comment]
We will not plan to do this work for the current release (but potentially hand to another developer after the upcoming new version is out the door). We will briefly discuss at design meeting.

**Comment by jbphet** [comment]
I believe that the code changes could be confined to the `MagnetFieldLines` class, and would either involve changing the positions of the ellipses or using elliptical arcs instead.  We will also probably need to add a line that goes through the center.

We discussed this in a design meeting on 10/29/2020, and @arouinfar is going to do a mockup that will specify how we want it to look, since there are several examples above and they are not all consistent with one another.

**Comment by arouinfar** [comment]
Here's a mockup of what the updated field lines could look like (please pretend the arrows aren't misisng). The spacing between each loop is similar and the lines don't touch at the magnet's poles. 
![image](https://user-images.githubusercontent.com/8419308/105262536-66308f80-5b4d-11eb-9054-8ae2d9f95366.png)
![image](https://user-images.githubusercontent.com/8419308/105262911-7cd6e680-5b4d-11eb-9bdb-509c4ff49e41.png)

Side-by-side comparison with master:
![image](https://user-images.githubusercontent.com/8419308/105262308-4c8f4800-5b4d-11eb-8c93-26968be9e853.png)
